src/libraries/System.Threading.Tasks.Parallel/src/System/Threading/Tasks/Parallel.ForEachAsync.cs

```diff
@@ -93,17 +93,23 @@ private static Task ForEachAsync<TSource>(IEnumerable<TSource> source, int dop,
                     // Continue to loop while there are more elements to be processed.
                     while (!state.Cancellation.IsCancellationRequested)
```

Comment:
Does checking the `IsCancellationRequested` here add any value, compared to just doing `while (true)`? The check is performed after acquiring the lock anyway.

Reply:
It's six-of-one, half-dozen-of-another.  The check is cheap and saves doing a lot of unnecessary work in the case of cancellation having been requested.  Since it doesn't meaningfully hurt the common case, I think it's fine to keep it.